packages/bsky/src/services/indexing/index.ts

```diff
@@ -93,6 +103,57 @@ export class IndexingService {
     }
   }
 
+  async indexRepo(did: string, commit: string) {
+    const now = new Date().toISOString()
+    const { pds, signingKey } = await this.didResolver.resolveAtpData(did)
+    const { sync } = new AtpAgent({ service: pds }).api.com.atproto
+
+    const { data: car } = await sync.getCheckout({ did, commit })
+    const { root, blocks } = await readCarWithRoot(new Uint8Array(car))
+    const storage = new MemoryBlockstore(blocks)
+    const repo = await verifyCheckoutToRepo(storage, root, did, signingKey)
+
+    // First, replace the user with just their current profile if it exists
```

Comment:
is there a reason for handling profile before the rest of the records? & in a different tx as well?

Reply:
It's really just a UX thing.  When we start reindexing the user their data will disappear then gradually flow back in.  I figure it would be friendly to ensure that their profile is immediately up to date since it's sort of the "face" of that user.

Reply:
ah right yup that makes sense 👍